src/FolderContext.ts

```diff
@@ -23,10 +24,11 @@ export class FolderContext implements vscode.Disposable {
 	private constructor(
         public folder: vscode.WorkspaceFolder,
         public swiftPackage: SwiftPackage,
-        readonly isRootFolder: boolean
+        readonly isRootFolder: boolean,
+        workspaceContext: WorkspaceContext
```

Comment:
We could upgrade this to a property. That way, we don't have to pass both the folder context and the workspace context to PackageWatcher.

Reply:
That'd be a cyclical reference

Reply:
We already have two cycles here I think:

`FolderContext` <-> `PackageWatcher`
`WorkspaceContext` -> `FolderContext -> `PackageWatcher`-> `WorkspaceContext`

We should double-check how we store our state to make sure the mark-and-sweep GC can clean everything up.

Reply:
I'll add another PR to clean the cycle up

Reply:
So this is good to go then?